clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3257,6 +3388,13 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
   KernelObjVisitor Visitor{*this};
   Visitor.VisitRecordBases(KernelObj, kernel_decl, kernel_body, int_header);
   Visitor.VisitRecordFields(KernelObj, kernel_decl, kernel_body, int_header);
+
+  ParmVarDecl *KernelHandlerArg = getSyclKernelHandlerArg(KernelCallerFunc);
```

Comment:
I'd suggest just combining this and the 'if' below (that is, do the init the 'if').